lan867x/idf_component.yml

```diff
@@ -1,6 +1,4 @@
 version: "0.9.0"
-targets:
-  - esp32
```

Comment:
@bogdankolendovskyy please remind me, why we are removing it? Only ESP32 and ESP32P4 have RMII.

Reply:
Since it is dependency of `ethernet_init` it makes `ethernet_init`'s only supported target be esp32 ([link to comment](https://github.com/espressif/esp-eth-drivers/pull/23#discussion_r1536560704)).

Reply:
I see, thanks for the reminder. The ideal option would be to update `ethernet_init` manifest file with [matching rules](https://docs.espressif.com/projects/idf-component-manager/en/latest/reference/manifest_file.html#matches-and-rules) as follows:
```
dependencies:
  idf:
    version: '>=5.1'
  lan867x:
    version: '*'
    rules:
     - if: "target in [esp32]"
 ```

Unfortunately, there is some issue with current version of Component Registry manager and dependencies are process prior rules. There is fix planned but it will still take some time. Therefore, the most safe approach is to remove the LAN867x from the `ethernet_init` component and create a JIRA task to return it once the Component manager is fixed. We cannot remove ESP32 target dependency from LAN876x because it's really dependent on this target due to RMII.

Reply:
Done. lan867x dependency removed from `ethernet_init` and `target` option is restored in lan867x `idf_component.yml`

Reply:
Please also update `Kconfig.projbuild`

```
            config ETHERNET_PHY_LAN867X
                # TODO: un-hide this option once processing of Matches and Rules is fixed in  IDF Component Manager and so LAN867x component can be conditionally included. 
                bool
                help
```

Please also perform additional updates to make to workflow passes.